tests/test_terra_notebook_utils.py

```diff
@@ -86,5 +90,49 @@ def test_extract(self):
                 self.assertGreater(time.time() - start_time, age)
 
 
+class TestTerraNotebookUtilsProgress(unittest.TestCase):
+    def test_progress_reporter(self):
+        with progress.ProgressReporter() as pr:
+            pr.checkpoint(2)
+            pr.checkpoint(5)
+            pr.checkpoint(7)
+            time.sleep(2)
+            self.assertEqual(14, pr.units_processed)
+
+    def test_rate_limited(self):
+        rate = 50
+        with self.subTest("Should be rate limited to expected number of calls"):
+            call_info = dict(number_of_calls=0)
+
+            @progress.RateLimited(rate)
+            def rate_limited_func():
+                call_info['number_of_calls'] += 1
+
+            start_time = time.time()
+            duration = 0
+            while duration <= 4.5 * 1 / rate:
```

Comment:
I'd expect this to be `5 * 1`?

Reply:
The first call is at `t=0`. If we waited for 5 seconds to ellapse I'd expect to get 6 calls, not 5.